source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -521,6 +521,25 @@ class CaresDnsResolverFactory : public DnsResolverFactory {
     }
     return std::make_shared<Network::DnsResolverImpl>(cares, dispatcher, resolvers);
   }
+
+  void init() override {
+    // Initialize c-ares library in case first time.
+    absl::Mutex mutex;
```

Comment:
this mutex is not going to do anything. It needs to be a class member.

Two different threads could call init() concurrently and they'd each get separate local mutexes.